nuxeo-features/nuxeo-automation/nuxeo-automation-server/src/main/java/org/nuxeo/ecm/automation/server/MarshallerDescriptor.java

```diff
@@ -26,24 +26,23 @@
 
 import org.nuxeo.common.xmap.annotation.XNodeList;
 import org.nuxeo.common.xmap.annotation.XObject;
+import org.nuxeo.runtime.model.Descriptor;
 
 /**
  * @since 5.8
  */
 @XObject("marshaller")
-public class MarshallerDescriptor {
+public class MarshallerDescriptor implements Descriptor {
 
     @XNodeList(value = "writer", componentType = Class.class, type = ArrayList.class)
-    private List<Class<? extends MessageBodyWriter<?>>> writers;
+    public List<Class<? extends MessageBodyWriter<?>>> writers = new ArrayList<>(0);
 
     @XNodeList(value = "reader", componentType = Class.class, type = ArrayList.class)
-    private List<Class<? extends MessageBodyReader<?>>> readers;
+    public List<Class<? extends MessageBodyReader<?>>> readers = new ArrayList<>(0);
 
-    public List<Class<? extends MessageBodyWriter<?>>> getWriters() {
-        return writers;
+    @Override
+    public String getId() {
+        return toString();
```

Comment:
I don't like this... but I don't see a much simpler way so let's keep it for now...